samples/performance/pipeline_barriers/pipeline_barriers_tutorial.md

```diff
@@ -0,0 +1,177 @@
+<!--
+- Copyright (c) 2019, Arm Limited and Contributors
+-
+- SPDX-License-Identifier: Apache-2.0
+-
+- Licensed under the Apache License, Version 2.0 the "License";
+- you may not use this file except in compliance with the License.
+- You may obtain a copy of the License at
+-
+-     http://www.apache.org/licenses/LICENSE-2.0
+-
+- Unless required by applicable law or agreed to in writing, software
+- distributed under the License is distributed on an "AS IS" BASIS,
+- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+- See the License for the specific language governing permissions and
+- limitations under the License.
+-
+-->
+
+# Using pipeline barriers efficiently
+
+## Overview
+
+Vulkan gives the application significant control over memory access for resources.
+Pipeline barriers are particularly convenient for synchronizing memory accesses between render passes.
+
+Having barriers is required whenever there is a memory dependency - the application should not assume
+that render passes are executed in order.
+
+However, having too many or too strict barriers can affect the application's performance.
+This sample will cover how to set up pipeline barriers efficiently, with a focus on pipeline stages.
+
+## Tile-based rendering
```

Comment:
So describing tile renders is nothing new for anyone involved in mobile :laughing:, but if someone asked about tilers I wouldn't think of finding documentation in the Pipeline Barriers section of the Vulkan Samples. I feel that the description of tilers doesn't belong here. Like all GPU's are highly parallel machines and I think its not fair to word this section as if pipeline barriers don't matter for intermediate mode renders just as much as tilers.

This documentation should stick to talking about pipeline barriers being used for allowing a renderpass to not be stuck on a stage of work and not go down the tiler rabbit hole here

Reply:
I believe the section is geared to serve as a refresher to tile based rendering. However I'll let @AttilioProvenzano-ARM answer to this properly as he was the author.

Reply:
This section was meant as a refresher, not really a full explanation of how tilers work. We can reduce it if it gives that impression.

> Like all GPU's are highly parallel machines and I think its not fair to word this section as if pipeline barriers don't matter for intermediate mode renders just as much as tilers.
Agreed, the reason for this section is giving motivation for pipeline barriers. The focus was on tilers as our project was originally focused on mobile. I'll rewrite this section to be more general.